src/Runtime/Distributions/SequenceDistribution.cs

```diff
@@ -167,26 +170,13 @@ public static TThis Zero()
             return result;
         }
 
-        /// <summary>
-        /// Creates a distribution from a given weight (non-normalized probability) function.
-        /// </summary>
-        /// <param name="sequenceToWeight">The weight function specifying the distribution.</param>
-        /// <returns>The created distribution.</returns>
-        [Construction(nameof(GetWorkspaceOrPoint), UseWhen = nameof(UsesAutomatonRepresentation))]
-        public static TThis FromWeightFunction(TAutomaton sequenceToWeight)
-        {
-            Argument.CheckIfNotNull(sequenceToWeight, nameof(sequenceToWeight));
-
-            return FromWorkspace(sequenceToWeight.Clone());
-        }
-
         /// <summary>
         /// Creates a distribution from a given weight (non-normalized probability) function.
         /// </summary>
         /// <param name="sequenceToWeight">The weight function specifying the distribution.</param>
         /// <returns>The created distribution.</returns>
         [Construction(nameof(GetWeightFunction))]
-        public static TThis FromWeightFunction(TWeightFunction sequenceToWeight)
+        public static TThis FromWorkspace(TWeightFunction sequenceToWeight)
```

Comment:
Why change the name?

Reply:
Oh, thanks for drawing attention here. Before the introduction of `IWeightFunction` there were `(From|Set)WeightFunction` and `(From|Set)Workspace` methods. All were accepting an automaton as an argument. In #339 I added corresponding overloads that would accept a weight fuction of type `TWeightFunction` (which usually is a multirepresentable weight fucntion).

The difference between `*WeightFunction` and `*Workspace` methods was that the former cloned the argument, while the latter didn't, so calling this method that did not clone its argument `FromWeightFunction` was a mistake already.

This PR makes weight functions constant and thread safe (the only mutation allowed is the lazy evaluation of flags on automata), which obsoletes methods that clone the argument. So I just removed those and fixed the name for this method.

Now, since the weight functions became immutable, it's now impossible to change the distribution via a reference to its weight function, so referring to them as 'workspaces' in API sounds strange. So, maybe, it would be better to use `(From|Set)WeightFunction` names?

Reply:
I agree that workspace is not the right term anymore.